### PR TITLE
fix(core): reorganized package structure

### LINKED ISSUE #876: Twig: Reorganize how components are exported in the package
This extends [#762](https://github.com/international-labour-organization/designsystem/issues/762)

Each Twig component should include:

- Twig file
- Pattern file (yml)
- Bundled behaviour file (.js)
- Stylesheet for that component

This needs to be done in a way that supports the legacy implementation (where assets are retrieved directy from `/src/`)

### PR DESCRIPTION
**Pull request by bashlk**
Fixes #876 and #874 

Previously the dist folder was not being published to npm and it didn't contain the twig / wingsuit components in a format that was ready to be consumed by Drupal.

This PR changes the structure of the dist folder such that the everything needed by Drupal to render a component is placed within its own folder. (i.e. the twig template, the wingsuit definition in yaml, the bundled javascript and the compiled css for a component are placed in a folder named after the component) So this also addresses #762 for the twig package since the compiled styles are now inside the twig package and the end user does not need to import and use the styles package seperately.

This PR also introduces a .npmignore file to make npm publish the dist folder and to avoid it from publishing the storybook and other dev tool configurations.

### CODE REVIEW

File: packages/twig/package.json [hunk 1]
```diff
@@ -41,15 +40,19 @@
     "@ilo-org/styles": "workspace:*",
     "@ilo-org/themes": "workspace:*",
     "@ilo-org/utils": "workspace:*",
+    "@popperjs/core": "^2.11.8",
     "video.js": "^7.21.2",
-    "videojs-youtube": "^3.0.1",
-    "@popperjs/core": "^2.11.8"
```
Comment (justintemps) [inline packages/twig/package.json:46]: @bashlk we should probably leave this in alphabetical order, no?

**Comment by justintemps** [comment]
Hey @bashlk, I just ran `npm pack` in the Twig directory and this is the structure I'm seeing:

```
├── CHANGELOG.md
├── LICENSE
├── README.md
├── directory-structure.txt
├── package.json
└── src
    ├── namespaces.js
    ├── patterns
    │   └── components
    ├── tokens
    │   ├── exports
    │   └── index.scss
    └── vendorjs
        └── lib.vendor.js
```

File: packages/twig/package.json (hunk 1 above)
Comment (bashlk) [reply packages/twig/package.json:46]: I didn't rearrange the order of package, pnpm did 😄

**Comment by bashlk** [comment]
@justintemps Apparently the `npm pack` command doesn't work with pnpm packages. We should run `pnpm pack` instead. Changesets apparently uses `pnpm pack` automatically when it detects it - https://github.com/changesets/changesets/blob/main/docs/command-line-options.md#publish

**Comment by justintemps** [comment]
@bashlk that's correct, but even with `pnpm pack` I'm not seeing the `dist` folder.

**Comment by justintemps** [comment]
@bashlk, I add `!dist` to the `.npmignore` file in the Twig package and that seemed to work. I think `pnpm pack` might check both the `.npmignore` and `.gitignore` files and if a file is specified in either one, then it won't added it to the tarball.

File: packages/twig/package.json (hunk 1 above)
Comment (justintemps) [reply packages/twig/package.json:46]: That's really weird. I'm not going crazy, `@` packages should come first right?

**Comment by ademarco** [comment]
@bashlk not sure it's related to the current PR but it didn't seem to happen on develop. When I build the design system I get the css of some components in their own directory, and not alongside the component:

image

Maybe this is due to the fact that the name of the component, in the YML file, (`datacard`) does not match the name of the component file and directory (`card_data`)? There are several such cases.

**Comment by justintemps** [comment]
hey @bashlk I just pushed up a few commits that 

1. Remove all of the `.npmrc` files throughout the project
2. Add a `files` property to the root package.json

This has the desired effect of packing the modules correctly with all of the `dist` assets that we need. Have a look and give it a try to make sure you agree with what I've included.

⚠️ I did not include the Twig `/src` folder. I checked the ILO.org Drupal project and the `/src` path is only getting used in two places, so I think we can easily make a PR to that project when we're ready with a new version of the Design System and the correct path for the Twig assets.

**Comment by bashlk** [comment]
@justintemps The current wildcard glob patterns for the LICENSE, README.md and more are also matching files inside the node_modules folders of the individual packages. We shouldn't include node_modules in the published assets. Also the package.json files for all of the packages are missing.

I think we should also remove the useless files properties inside each packages' package.json file. I already had this change locally so I pushed it up just now.

**Comment by justintemps** [comment]
@bashlk good catch. I'll finish working on the packages on this branch since we're already here. I'll then leave you to make any changes to the Twig package.

**Comment by bashlk** [comment]
@ademarco Yes, the styles are being placed in the wrong folder because the filenames don't match (See copystyles.js) - previously all of the CSS files were being placed in a single folder, I changed it so that they are being placed alongside the other component files. Do you think the previous approach was better?

**Comment by ademarco** [comment]
> @ademarco Yes, the styles are being placed in the wrong folder because the filenames don't match (See copystyles.js) - previously all of the CSS files were being placed in a single folder, I changed it so that they are being placed alongside the other component files. Do you think the previous approach was better?

We can actually roll back to the previous approach, as I realized that the current wingsuit definition do not properly declare dependencies between patterns and, in failing to do so, we are forced to load the CSS of all components, or styles will break.

If possible, I would keep the new approach of having the JS compiled in .behaviour.js alongside the twig files tho, as those can be loaded conditionally.

**Comment by justintemps** [comment]
@bashlk ok, new glob pattern in the root package.json adds package.json and removes node_modules. All the packages are now significantly smaller. 🎉

**Pull request merged**